cmake/ccache.cmake

```diff
@@ -17,12 +19,23 @@ if(NOT MSVC)
       )
       set(WITH_CCACHE ON)
     elseif(WITH_CCACHE)
-      list(APPEND CMAKE_C_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
-      list(APPEND CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
+      foreach(lang IN ITEMS C CXX OBJCXX)
+        if(DEFINED CMAKE_${lang}_COMPILER_LAUNCHER)
+          list(APPEND configure_warnings
+            "CMAKE_${lang}_COMPILER_LAUNCHER is already set. Skipping built-in ccache support for ${lang}."
+          )
```

Comment:
Is making this a `configure_warning` compatible with #31665? Maybe it would be a benefit if CI complained when unintentionally skipping ccache.

Reply:
Would you mind elaborating on your suggestion?

Reply:
Maybe better expressed as: Is trying to both override the compiler launcher and make configure_warnings errors a use case that should be supported? If so, making this message a `configure_warning` is incompatible with #31665 since that makes `configure_warnings` fatal errors if `WCONFIGURE_ERROR=ON`.

If it is not a use case that should be supported, then making this message a configure_warning might be good together with #31665, since it would cause an error in CI if ccache ever got disabled because a compiler launcher was set, but maybe there isn't a realistic situation where something like that could happen unintentionally.

Reply:
It's also fair if you just think #31665 is a bad approach and this is more evidence of that.